### PR TITLE
fixing broken build

### PR DESCRIPTION
**Pull request by dmitryv**
Fix for ( ruby 1.9.3p448 ) :

  1) Failure:
TracerouteTest#test_defined_action_methods [/Users/dd6/work/ht/traceroute/test/traceroute_test.rb:10]:
--- expected
+++ actual
@@ -1 +1 @@
-["users#index", "users#index2", "users#show", "admin/shops#index", "admin/shops#create"]
+["users#index2", "users#show", "users#index", "admin/shops#create", "admin/shops#index"]

### CODE REVIEW

**Comment by amatsuda** [comment]
@dmitryv Thank you for the fix! But can you please configure your Git not to track `.idea/` thing and then `push -f` you commit here?

File: lib/traceroute.rb [hunk 1]
```diff
@@ -20,26 +20,44 @@ def load_everything!
...
       end
     end.reject {|r| r.start_with? 'rails/'}
+
+    routed_actions = routed_actions - exclusions unless exclusions.nil?
+
+    routed_actions
+  end
+
+  def self.load_exclusions(file_name)
+    return if file_name.nil?
```
Comment (torrocus) [inline lib/traceroute.rb]: `return unless file_name`

File: lib/traceroute.rb [hunk 2]
```diff
@@ -20,26 +20,44 @@ def load_everything!
...
-  def defined_action_methods
-    ActionController::Base.descendants.map do |controller|
+  def defined_action_methods(exclusions = nil)
+    routes = ActionController::Base.descendants.map do |controller|
       controller.action_methods.reject {|a| (a =~ /\A(_conditional)?_callback_/) || (a == '_layout_from_proc')}.map do |action|
         "#{controller.controller_path}##{action}"
       end
     end.flatten.reject {|r| r.start_with? 'rails/'}
+
+    routes = routes - exclusions unless exclusions.nil?
```
Comment (torrocus) [inline lib/traceroute.rb]: `routes -= exclusions if exclusions`

File: lib/traceroute.rb [hunk 3]
```diff
@@ -20,26 +20,44 @@ def load_everything!
...
+  def routed_actions(exclusions = nil)
+    routed_actions = routes.map do |r|
       if r.requirements[:controller].blank? && r.requirements[:action].blank? && (r.path == '/:controller(/:action(/:id(.:format)))')
         %Q["#{r.path}"  This is a legacy wild controller route that's not recommended for RESTful applications.]
       else
         "#{r.requirements[:controller]}##{r.requirements[:action]}"
       end
     end.reject {|r| r.start_with? 'rails/'}
+
+    routed_actions = routed_actions - exclusions unless exclusions.nil?
```
Comment (torrocus) [inline lib/traceroute.rb]: `routed_actions -= exclusions if exclusions`

File: lib/traceroute.rb [hunk 4]
```diff
@@ -20,26 +20,44 @@ def load_everything!
...
+
+    routed_actions
+  end
+
+  def self.load_exclusions(file_name)
+    return if file_name.nil?
+    exclusions = []
+    File.open(file_name, 'r'){|f|
```
Comment (torrocus) [inline lib/traceroute.rb]: `exclusions = File.readlines(file_name).map(&:strip).select(&:present?)`

**Pull request closed without merging**